### PR TITLE
feat: meta desc iterations

### PR DESCRIPTION
**Pull request by fairlighteth**
- Iterates on Meta title and description
- Picks from a randomized template list for generating meta tag descriptions
- Adds dynamic 🟢 or 🔴 and up/down arrows in the meta title.
- Add additional meta tag overrides (canonical, og:url, etc.)
- Fixed issue with not properly overriding the `<Layout>` meta tags. Added a `key` attribute conform Next.js docs to have unique tags and no duplicates.

Test meta tags here https://technicalseo.com/tools/google-serp-simulator/

## SERP preview
Screenshot 2023-07-06 at 15 56 54
Screenshot 2023-07-06 at 15 57 18

Screenshot 2023-07-06 at 15 58 13

### CODE REVIEW

File: pages/tokens/[id].tsx [hunk 1]
```diff
@@ -8,18 +8,30 @@ import { CONFIG } from '@/const/meta'
...
+}
 
-export default function TokenDetailsPage({ token }: TokenDetailPageProps) {
-  const { name, symbol, metaDescription } = token
+export default function TokenDetailsPage({ token, currentUrl }: TokenDetailPageProps) {
+  const { name, symbol, metaDescription, change24h, priceUsd } = token
+  const change24 = parseFloat(change24h)
+  const change24hFormatted = change24.toFixed(2)
+  const isIncrease = parseFloat(change24h) >= 0;
+  const changePlusMinus = isIncrease ? '+' : '-';
```
Comment (alfetopito) [inline pages/tokens/[id].tsx:20]: You might not want to display the `-` as the number itself will have the signal when negative:

image

Comment (fairlighteth) [reply pages/tokens/[id].tsx]: Removed, but now for me doesn't show the minus or plus.

Comment (alfetopito) [reply pages/tokens/[id].tsx]: Fixed for page title, meta description still has double `-`

image

Comment (fairlighteth) [reply pages/tokens/[id].tsx]: @alfetopito Removed, but again, for me now there is no + or -. Not sure why for you it's doable. As if code is rendered differently.

**Review by alfetopito — APPROVED** [review]

File: pages/tokens/[id].tsx [hunk 2]
```diff
@@ -11,15 +11,22 @@ const DATA_CACHE_TIME_SECONDS = 10 * 60 // 10 minutes
 export type TokenDetailPageProps = TokenDetailProps
 
 export default function TokenDetailsPage({ token }: TokenDetailPageProps) {
-  const { name, symbol, metaDescription } = token
+  const { name, symbol, metaDescription, change24h, priceUsd } = token
+  const change24 = parseFloat(change24h)
+  const change24hFormatted = change24.toFixed(2)
+  const isIncrease = parseFloat(change24h) >= 0;
+  const priceChangeEmoji = isIncrease ? '🟢' : '🔴';
```
Comment (anxolin) [inline pages/tokens/[id].tsx:18]: cool

File: services/tokens/index.ts [hunk 3]
```diff
@@ -120,22 +120,34 @@ function _getTokenDescription(id: string): string {
...
+    `Interested in ${name} (${symbol})? Priced at $${priceUsd}, with ${changeDirection} of ${change24hTrimmed}% in 24h. Trading volume: $${volume}. Their market cap: $${marketCap}. Learn more about ${symbol}!`,
+    `Graze on this: ${name} (${symbol}) at $${priceUsd}, with ${changeDirection} of ${change24hTrimmed}% in 24h. Trading volume: $${volume}. They boast a market cap of $${marketCap}. Learn about ${symbol}.`,
+    `Ever heard of ${name} (${symbol})? At $${priceUsd}, they've marked a ${changeDirection} of ${change24hTrimmed}% in 24h. Trading volume: $${volume}. Their market cap: $${marketCap}. Get to know them.`,
+    `Check out ${name} (${symbol})! Grazing at $${priceUsd}, with ${changeDirection} of ${change24hTrimmed}% in 24h. Trading volume: $${volume}. Their market cap: $${marketCap}. Discover ${symbol}'s secrets.`,
+    `Latest on ${name} (${symbol}): priced at $${priceUsd}. Experienced a ${changeDirection} of ${change24hTrimmed}% in 24h. Trading volume: $${volume}. Their market cap: $${marketCap}. Learn more.`,
+  ];  
+  
+  const randomIndex = Math.floor(Math.random() * templates.length);
```
Comment (anxolin) [inline services/tokens/index.ts:143]: you generate all templates in order to generate one message

**Review by anxolin — COMMENTED** [review]
Great PR, one small comment on the random templates

File: pages/tokens/[id].tsx (hunk 1 above)
Comment (alfetopito) [reply pages/tokens/[id].tsx]: Yeah, you removed both, and I'm fine with that.

Now, only negative values have the sign
image
image

**Pull request merged**